### PR TITLE
#21916 setting a default password to admin

### LINKED ISSUE #21916: Make initial admin password configurable
1. When a new dotCMS instance is first started up (starter.zip import), the system should auto-generate a non-guessable password for the admin@dotcms.com user.
2. We need to allow for a configuration option `INITIAL_ADMIN_PASSWORD` that, if set, will override the auto-generated password and preset the admin@dotcms.com user's password.
3.  Finally, whatever the new admin password is, we spit the value out into the logs in a very clear way, e.g.
``` 
#######
#######    NOTICE: admin@dotcms.com password set to 'XXXXXXXXXXX'
#######
```

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This will set a new generated secure password to the default admin user
Then the password gets printed out to the logs. This only happens the first time right after loading the starter.
Also there's the possibility to set up a predetermined password loaded via properties

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java [hunk 1]
```diff
@@ -135,7 +142,41 @@ private static void removeAnyOldMetadata() throws DotDataException {
                String.format(" `%d` old metadata entries removed and `%d` old content files from the starter." , tuple._1 , + tuple._2)
             );
         }catch (ExecutionException | InterruptedException e){
-            Logger.error(DotCMSInitDb.class,"");
+            Logger.error(DotCMSInitDb.class,"An error occurred removing old metadata ",e);
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java:145]: did you check if this message becomes so noisy? You changed it to print the stack trace also

File: dotCMS/src/main/java/com/dotmarketing/util/UserUtils.java [hunk 2]
```diff
@@ -20,4 +20,16 @@ public static List getChallengeQuestionList() {
...
+
+	static final String ACCEPTABLE_PASSWORD_CHARS = "!#%+23456789:=?@ABCDEFGHJKLMNPRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+
+	/**
+	 * Generates a passwords that might be any random combination of 16 chars lenght
+	 * with any of these chars !#%+23456789:=?@ABCDEFGHJKLMNPRSTUVWXYZabcdefghijkmnopqrstuvwxyz
+	 * @return
+	 */
+	public static String generateSecurePassword(){
+	   	return RandomStringUtils.random(16, ACCEPTABLE_PASSWORD_CHARS.toCharArray());
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/util/UserUtils.java:32]: We will need at least 1 special char, one number, one uppercase and one lowercase in every generated password.

File: dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java [hunk 3]
```diff
@@ -135,7 +142,41 @@ private static void removeAnyOldMetadata() throws DotDataException {
...
+        printNotice(message);
+    }
+
+    /**
+     * inform user the default password set to the admin user
+     * @param message the notice message
+     */
+    static void printNotice(final String message){
+        final int width = 100;
+        Logger.info(DotCMSInitDb.class,StringUtils.rightPad("#", width, "#"));
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java:178]: Nice

File: dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/db/DotCMSInitDb.java:145]: it's not noisy at all

**Review by jcastro-dotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**